Sources/SwiftFusion/Inference/NewJacobianFactor.swift

```diff
@@ -66,11 +66,25 @@ public struct NewJacobianFactor<
   }
 
   public func errorVector_linearComponent(_ x: Variables) -> ErrorVector {
-    return ErrorVector(jacobian.lazy.map { $0.dot(x) })
+    var r = ErrorVector.zero
+    jacobian.withUnsafeBufferPointer { jacRows in
+      r.withUnsafeMutableBufferPointer { rScalars in
+        for i in 0..<ErrorVector.dimension {
+          rScalars[i] = jacRows[i].dot(x)
+        }
+      }
+    }
+    return r
   }
 
   public func errorVector_linearComponent_adjoint(_ y: ErrorVector) -> Variables {
-    return zip(y.scalars, jacobian).lazy.map(*).reduce(Variables.zero, +)
+    y.withUnsafeBufferPointer { yScalars in
+      jacobian.withUnsafeBufferPointer { jacRows in
+        (0..<ErrorVector.dimension).reduce(into: Variables.zero) { (result, i) in
+          result += yScalars[i] * jacRows[i]
```

Comment:
Can't this be expressed as some kind of `dot`?

Reply:
If we had some way of accessing the columns of the jacobian, then it could be a dot:
```
Variables(jacobian.columns.lazy.map { col in col.dot(y) })
```

But writing the `.columns` collection seems like complicated overkill and I expect it also wouldn't optimize well.

We can't simply switch `jacobian` to column-major because `errorVector_linearComponent` needs to access the rows.

Reply:
We need .columns for usability anyway, and I don't know why you think it wouldn't optimize well; maybe it would, maybe not, but it's worth doing.

Reply:
I would like to defer `.columns` until later because it's pretty non-trivial to "calculate" the type of `.columns`.

The rows are fixed-size arrays of tuple of vectors: types like `Array6<Tuple2<Vector3, Vector3>>`. I believe there's no way to "transpose" these types to the column types like `Tuple2<Array3<Vector6>, Array3<Vector6>>`.

The right solution is probably to define fixed-size matrices that know their "rows" type and their "columns" type. But there is a lot of fixed-size matrix design that we would need to do that is out of scope of this PR.